crates/services/sync/src/import.rs

```diff
@@ -186,6 +215,9 @@ where
                         return Ok(None)
                     }
 
+                    // Wait for the da to be at least the da height on the header.
+                    await_at_least_synced(relayer.as_ref(), &header.entity.da_height, &params.max_da_lag, params.max_wait_time).await?;
```

Comment:
You need to do it before `check_sealed_header`.

Also, I guess it is the responsibility of the `BlockVerifier` to await `Relaeyr`.
You don't need to pass `Relayer` to the `fuel-core-sync`. The method `check_sealed_header` should be async and here you need to call it with a timeout.

Reply:
Also if we really need to move it down to verify then we can do that in a follow up because it will also break the producer which isn't great

Reply:
To verify the signature, we need access to the relayer and logs.

`BlockProducer` doesn't use `BlockVerifeir`, so it should not affect him

Reply:
Why do we need logs to verify the signature?

Reply:
Because the `Relayer` is our source of the truth about block producers(But it depends on our key rotation strategy).

Reply:
Ah I see. Ok will make this change next week. I still think the signature check should be before awaiting the relayer though. Even if it's all done within the verifier. We should at least check the signature is correct for the header so we aren't awaiting for a da_height that no one has signed.